Explorer/Assets/DCL/AvatarRendering/Emotes/Systems/Play/EmotePlayer.cs

```diff
@@ -164,27 +178,55 @@ private static EmoteReferences CreateNewEmoteReference(GameObject mainAsset)
                 }
             }
 
-            references.Initialize(avatarClip, propClip, animator, propClipHash);
+            references.Initialize(avatarClip, propClip, animatorComp, animationComp, propClipHash);
 
             ListPool<AnimationClip>.Release(uniqueClips);
 
             // some of our legacy emotes have unity events that we are not handling, so we disable that system to avoid further errors
-            if (animator != null)
-                animator.fireEvents = false;
+            if (animatorComp != null)
+                animatorComp.fireEvents = false;
 
             return references;
         }
 
-        private void PlayLegacyEmote(GameObject avatarAnimatorGameObject, AnimationClip legacyAnimationClip, bool loop)
+        private void PlayLegacyEmote(GameObject avatarAnimatorGameObject, ref CharacterEmoteComponent emoteComponent, EmoteReferences emoteReferences, bool loop)
         {
             Animation animationComp;
             if (!(animationComp = avatarAnimatorGameObject.GetComponent<Animation>()))
                 animationComp = avatarAnimatorGameObject.AddComponent<Animation>();
+            ClearLegacyAnimationClips(animationComp);
             animationComp.playAutomatically = false;
 
-            legacyAnimationClip.wrapMode = loop ? WrapMode.Loop : WrapMode.Once;
-            animationComp.AddClip(legacyAnimationClip, legacyAnimationClip.name);
-            animationComp.Play(legacyAnimationClip.name);
+            if (emoteReferences.avatarClip != null)
+            {
+                emoteComponent.EmoteLoop = loop;
+
+                var avatarClipName = emoteReferences.avatarClip.name;
+                animationComp.AddClip(emoteReferences.avatarClip, avatarClipName);
+                animationComp[avatarClipName].wrapMode = loop ? WrapMode.Loop : WrapMode.Once;
+                animationComp.Play(avatarClipName);
+            }
+
+            if (emoteReferences.propClip != null && emoteReferences.animationComp != null)
+            {
+                var propAnimationComp = emoteReferences.animationComp;
+                var propClipName = emoteReferences.propClip.name;
+                propAnimationComp[propClipName].wrapMode = loop ? WrapMode.Loop : WrapMode.Once;
+                propAnimationComp.Play(propClipName);
+            }
+        }
+
+        private void ClearLegacyAnimationClips(Animation animationComp)
+        {
+            HashSet<string> animationsToRemove = new HashSet<string>();
```

Comment:
Could we use it from a pool or pre-allocate it?

Reply:
good catch, that is actually completely unneeded since the new legacy animation will always be overriding the previous one.

Tackled at: https://github.com/decentraland/unity-explorer/pull/4214/commits/782fe61832baaabd0a630b8a8be38d5d6cfce99b